master/internal/db/postgres_test_utils.go

```diff
@@ -319,6 +322,54 @@ func MockModelCheckpoint(
 	return ckpt
 }
 
+// MockModelCheckpointSteps returns a mock model checkpoint with specified steps.
+func MockModelCheckpointSteps(
+	ckptUUID uuid.UUID, tr *model.Trial, a *model.Allocation, stepsCompleted int,
+) model.CheckpointV2 {
```

Comment:
could add functional opts to `MockModelCheckpoint` instead and save writing an entire new function